lib/llm/src/local_model.rs

```diff
@@ -332,6 +340,25 @@ impl LocalModel {
             .await
     }
 
+    pub async fn register_runtime_config(
+        &mut self,
+        endpoint: &Endpoint,
+        runtime_config: ModelRuntimeConfig,
+    ) -> anyhow::Result<()> {
+        self.card.register_runtime_config(runtime_config);
+
+        // Update the card in etcd if we're attached
+        if let Some(etcd_client) = endpoint.drt().etcd_client() {
+            let kvstore: Box<dyn KeyValueStore> = Box::new(EtcdStorage::new(etcd_client.clone()));
+            let card_store = Arc::new(KeyValueStoreManager::new(kvstore));
+            let key = self.card.slug().to_string();
+            card_store
+                .publish(model_card::ROOT_PATH, None, &key, &mut self.card)
+                .await?;
+        }
```

Comment:
The frontend fetches the card from NATS when it first discovers the model (typically as soon as the worker is ready). It does not later update that card. The frontend hence won't see the changes in the card. I don't know if that's relevant to you here.

Reply:
This is good to know, I will refactor the code to include the runtime info possibly in the `LocalModel` as @PeaBrane suggested